### DISCUSSION TITLE
Field-level permissions

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#8154] [Ideas] [answered]
> Created by: Visiluna (pjones@visiluna.com) on 2015/06/16 22:04:13 +0000
> Votes at time of UserVoice import: 8

-----------------------------------

I'd like to be able to restrict certain fields to specific member groups. 

Example: A user profile that contains a bunch of fields for contact information and a field for a "commission rate." A user should be able to edit the fields for their contact information, but only a member of a "management" user group should be able to see/edit the "commission rate" field.

**Comment by angrybrad** [comment]
> Posted by Visiluna (pjones@visiluna.com) on 2015/06/16 22:04:13 +0000
-----------------------------------

An example of a comparable implementation is how ExpressionEngine allows the creation of different Form Layouts for different user groups. I think the method of setting access at the field level works better in my mind, but either approach would achieve the desire result.

**Comment by moacode** [comment]
@angrybrad I've written a plugin which should do what you're after (although we're a couple of years on!): https://plugins.craftcms.com/craft-fab-permissions

I'd be happy to hear any feedback you might have!

**Comment by brandonkelly** [comment, marked as answer]
This has been added for Craft 4, by way of conditional fields (and tabs) that support conditions on the logged-in user 🎉  (#10300)
